tsl/src/remote/txn.c

```diff
@@ -152,13 +153,17 @@ remote_txn_begin(RemoteTxn *entry, int curlevel)
 	}
 }
 
+/*
+ * Check if the access node transaction which is driving the 2PC on the datanodes is
+ * still in progress.
+ */
 bool
-remote_txn_is_still_in_progress(TransactionId access_node_xid)
+remote_txn_is_still_in_progress_on_access_node(TransactionId access_node_xid)
 {
 	if (TransactionIdIsCurrentTransactionId(access_node_xid))
 		elog(ERROR, "checking if a commit is still in progress on same txn");
 
-	return XidInMVCCSnapshot(access_node_xid, GetTransactionSnapshot());
+	return TransactionIdIsInProgress(access_node_xid);
```

Comment:
While this seems to be the right function to use, what are the differences to how things were before with `XidInMVCCSnapshot`?

Reply:
As @akuzm pointed out, `TransactionIdIsInProgress` reports the liveness of the XID across the entire set of ongoing txns whereas `XidInMVCCSnapshot` just checks against our current transaction snapshot here. The former might be a little bit more safer.

Reply:
I just want to make sure that we cover the case when the other transaction completes before the end of the "current" transaction, but other transaction is not visible in the snapshot.

So, let's say T_a starts before T_b but T_b commits before T_a. Then there's time between T_b commit and T_a commit where this function returns false. Do we handle that situation correctly?

Reply:
Returns `false` for `T_b`? No, if we use `TransactionIdIsInProgress` then it will not return `false` for `T_b` AIUI